### ISSUE TITLE
Import crashes

### ISSUE THREAD

**Issue by garycourt** [syllant/bggcli#1]
I am moving my collection data from one account to another. The export worked fine, but the import failed with the following error:

```
[bggcli] Importing games for '********' account...
[bggcli] Authenticating... [done]
[bggcli] Importing 153 games...
[bggcli] [1/153] 101.1...  (already in collection)[bggcli] ERROR: Unexpected error while controlling the UI!
Either the web pages have changed and bggcli must be updated, or the site is down for maintenance.
Cause: Message:
Stacktrace:
    at FirefoxDriver.prototype.findElementInternal_ (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/driver-component.js:10659)
    at FirefoxDriver.prototype.findElement (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/driver-component.js:10668)
    at DelayedCommand.prototype.executeInternal_/h (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12534)
    at DelayedCommand.prototype.executeInternal_ (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12539)
    at DelayedCommand.prototype.execute/< (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12481)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/bggcli/util/csvreader.py", line 60, in iterate
    callback(row)
  File "/usr/local/lib/python2.7/dist-packages/bggcli/commands/collection_import.py", line 50, in <lambda>
    csv_reader.iterate(lambda row: game_page.update(row))
  File "/usr/local/lib/python2.7/dist-packages/bggcli/ui/gamepage.py", line 111, in update
    getattr(self, "fill_%s" % key)(value)
  File "/usr/local/lib/python2.7/dist-packages/bggcli/ui/gamepage.py", line 53, in _wrapper
    (By.XPATH, "//div[@class='select-free'][contains(@id, 'editversion')]")))
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/support/wait.py", line 80, in until
    raise TimeoutException(message, screen, stacktrace)
TimeoutException: Message:
Stacktrace:
    at FirefoxDriver.prototype.findElementInternal_ (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/driver-component.js:10659)
    at FirefoxDriver.prototype.findElement (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/driver-component.js:10668)
    at DelayedCommand.prototype.executeInternal_/h (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12534)
    at DelayedCommand.prototype.executeInternal_ (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12539)
    at DelayedCommand.prototype.execute/< (file:///tmp/tmp1rmw0x/extensions/fxdriver@googlecode.com/components/command-processor.js:12481)

[bggcli] Close webdriver 'collection-import'
```

Does importing still work for you? (As in, did BGG change their page to break the tool? Or is it something on my end?)

**Comment by syllant** [comment]
Hi Gary,

Yes, as the message says, BGG has probably done some changes. I'll give a
look tonight and will try to update the script. I'll keep you in touch.

Sylvain

**Comment by syllant** [comment]
I've launched my end-2-end test (export-import with some test data), and all is still working fine on my side. The bug probably depends on some specific data. It occurs when filling the "Version" fields (predefined version of the game, or "Custom Version Info").

@garycourt Would you mind to provide the first 2 lines of the CSV file, i.e. with the header and the first game "101.1" (you might want to hide some private info before)?

**Comment by garycourt** [comment]
Certainly:

```
objectname,objectid,rating,own,fortrade,want,wanttobuy,wanttoplay,prevowned,preordered,wishlist,wishlistpriority,wishlistcomment,comment,conditiontext,haspartslist,wantpartslist,publisherid,imageid,year,language,other,pricepaid,pp_currency,currvalue,cv_currency,acquisitiondate,acquiredfrom,quantity,privatecomment,_versionid
"101.1","179180","0","1","0","0","0","1","0","0","0","","","","","","","","","","","","","","","","","","","","276391"
"12 Days of Christmas","170430","8","1","0","0","0","0","0","0","0","","","","","","","","","","","","","","","","","","","","261533"
```

**Comment by syllant** [comment]
This CSV file works well for me, so it's not about the data. I see 4 other possible explanations:
1. BGG was just experiencing an issue while you were importing you file. This is far from being uncommon with this site.
2. Firefox was not responsive enough during the operation. Here the problem occurs when filling data on the "Version" popup. The popup is opened first, then the fields are identified and filled. Should the fields not found after 5s, you will have this kind of errors. E.g. I have some integration tests on Sauce labs for which I have to increase this default timeout to 60s!
3. There is some differences in your environment that makes BGG delivering a different HTML for your case (unlikely).
4. You have done something on Firefox while it was running the test that could have change the expected behavior (very unlikely I guess)

My recommendations:
1. Please, just retry the operation to be sure it was not related to a temporary outage.
2. Otherwise, add "-c browser-keep=true" in your command line. This will prevent Firefox from closing at the end of the operation. You can verify whether the Version popup is opened or not. At the end, you can also copy the HTML code of the current page and attach it here so I can check you current page (again you might want to remove some private information). Also, you can define an environment variable "CI" with "true" as value (e.g. `export CI=true`) before launching the import. This will change the timeout for Firefox to 60s.

**Comment by garycourt** [comment]
Thanks for the awesome reply! 
I tried it again and it still failed the same. Setting `CI=true` throws new errors about `SAUCE_USERNAME` and others missing, so I undid that. I looked at the code and added a `.implicitly_wait(10)` to the Firefox webdriver, but I still get the same error. I'm also running Firefox headless on Linux, so I can't see the browser visually.

My suspicion is that there is a problem between my versions of Selenium and Firefox. I'm going to look into that further.
